Tests/SKCoreTests/BuildSystemManagerTests.swift

```diff
@@ -412,16 +404,19 @@ final class BuildSystemManagerTests: XCTestCase {
 // MARK: Helper Classes for Testing
 
 /// A simple `MainFilesProvider` that wraps a dictionary, for testing.
-private final class ManualMainFilesProvider: MainFilesProvider {
-  let lock: DispatchQueue = DispatchQueue(label: "\(ManualMainFilesProvider.self)-lock")
-  private var _mainFiles: [DocumentURI: Set<DocumentURI>] = [:]
-  var mainFiles: [DocumentURI: Set<DocumentURI>] {
-    get { lock.sync { _mainFiles } }
-    set { lock.sync { _mainFiles = newValue } }
+private final actor ManualMainFilesProvider: MainFilesProvider {
+  private var mainFilesDictionary: [DocumentURI: Set<DocumentURI>]
```

Comment:
One last comment: `Dictionary` is superfluous here because the type is already a dictionary. I would just name it `mainFiles`.

Reply:
The intention for the renaming is to avoid the confusion between `mainFiles: [DocumentURI: Set<DocumentURI>]`
`mainFiles: Set<DocumentURI>` https://github.com/apple/sourcekit-lsp/pull/917/files#diff-4e5dc27e47b431520ccbd13ad2e859517644e3437c73157b2193a7b14ec30570R414

it was a trade-off when i named it, but yeah the data type already convey itself, so let me rename it.

Reply:
I think that’s fine. `updateMainFiles` also is a really short methods and I don’t think it should cause too much confusion.